Util/src/main/java/io/deephaven/util/datastructures/intrusive/IntrusiveSoftLRU.java

```diff
@@ -176,36 +179,34 @@ private void touchSimple(@NotNull final T itemToTouch) {
     /**
      * Resize the cache by doubling, stopping at the maximum capacity. If maximum capacity is reached, then switch the
      * structure into LRU mode.
-     *
      */
     private void doResize() {
-        // Time to reallocate or change internal structures
-        final int newCapacity = Math.min(softReferences.length << 1, maxCapacity);
-        final SoftReference<T>[] oldRefs = softReferences;
-        // noinspection unchecked
-        softReferences = new SoftReference[newCapacity];
+        // First we will try to compact the references. Maybe we can get away without actually resizing
+        int nullsFound = 0;
+        for (int refIdx = 0; refIdx < softReferences.length; refIdx++) {
+            final T object = softReferences[refIdx].get();
+            if (object == null) {
+                nullsFound++;
+            } else if (nullsFound > 0) {
+                softReferences[refIdx - nullsFound] = softReferences[refIdx];
+                softReferences[refIdx] = getNull();
+                adapter.setSlot(object, refIdx - nullsFound);
+            }
+        }
 
-        // We have grown too large, switch to LRU mode.
+        // We free'd up space! Cool!
+        size -= nullsFound;
+        if (size < softReferences.length - 1) {
+            return;
+        }
```

Comment:
If you agree, delete line 192 and accept this suggestion, too.
```suggestion
        if (nullsFound > 0) {
            // We free'd up space! Cool!
            size -= nullsFound;
            Assert.lt(size, "size", softReferences.length, "softReferences.length");
            Arrays.fill(softReferences, size, softReferences.length, getNull());
            return;
        }
```